web/src/dropdown_widget.js

```diff
@@ -0,0 +1,134 @@
+import $ from "jquery";
+import * as tippy from "tippy.js";
+
+import render_dropdown_list from "../templates/dropdown_list.hbs";
+import render_dropdown_list_container from "../templates/dropdown_list_container.hbs";
+
+import * as ListWidget from "./list_widget";
+import {default_popover_props} from "./popover_menus";
+import {parse_html} from "./ui_util";
+
+/* Sync with max-height set in popovers.css */
+export const DEFAULT_DROPDOWN_MAX_HEIGHT = 200;
+
+export function setup(popover_props, get_options, item_click_callback) {
+    const on_show_callback = popover_props.on_show_callback;
+    // Since we pass `popover_props` to `tippy`, we should delete our custom properties.
+    delete popover_props.on_show_callback;
```

Comment:
I'm not sure, but it might be cleaner to just have a separate section for tippy props.

Reply:
Ok, makes sense.